warehouse/internal/repo/staging.go

```diff
@@ -329,24 +337,52 @@ func (repo *StagingFiles) Pending(ctx context.Context, sourceID, destinationID s
 			id DESC
 		LIMIT 1;
 	`, sourceID, destinationID,
-	).Scan(&lastStagingFileID)
+	).Scan(
+		&uploadID,
+		&lastStartStagingFileID,
+		&lastEndStagingFileID,
+		&useUploadID,
+	)
+
 	if err == sql.ErrNoRows {
-		lastStagingFileID = 0
+		lastEndStagingFileID = 0
```

Comment:
Should we reset `lastStartStagingFileID` as well?

Reply:
Wouldn't this value by default be `0` ? Given there are no rows so no data would be scanned from DB. Wdyt ?

Reply:
Should we even keep this check of `err == sql.ErrNoRows` ?

Reply:
It should have been removed indeed.